### PR TITLE
Fixing importStats.js to reduce DB Calls to prevent "Maximum call stack size exceeded" in docker-compose Instance

### PR DESCRIPTION
**Pull request by kejith**
Problem:
image

Running a docker-compose instance causes us to encounter this problem (at least for me). We start so many async DB calls that we risk going over the call stack limit and panicking mid-update.

Updates made consecutively would have had a significant performance impact, therefore I did not explore that solution.

The second option, which I put into practice, batch inserts `players_history` data and batch upserts `players` data into the database. decreasing the number of calls from `(2 * n)` times to `2`.

As a result of my current lack of confidence in my approach, I would strongly advise trying this locally before merging.

A unique index constraint must be added to the `player_id` column in the `players` table if you wish to utilize this with an existing database.

```sql
CREATE UNIQUE INDEX 
    players_player_id_index 
ON 
    "public"."players" 
    ( 
        "player_id" ASC 
    )
```

I adjusted setupDb.js for new configurations.

### CODE REVIEW

**Comment by kejith** [comment]
We should think about a database migration tool maybe?

**Comment by kejith** [comment]
Found a bug which leads to data inconsitencies. Trying to resolve it.

**Comment by kejith** [comment]
> Found a bug which leads to data inconsitencies. Trying to resolve it.

With the updates, I overwrote the player status. I believe that has been addressed, but I am now more skeptical of the validity of this PR.

File: lib/data/player.js [hunk 1]
```diff
@@ -80,6 +90,50 @@ module.exports = class Player {
...
   }
 
+  static async saveMany (players, tableName = 'players') {
+    logger.info('Save many')
+    const data = players.map((p) => {
+      const d = p.toDBformat()
+
+      delete d.is_banned
+      delete d.is_inactive
+      delete d.is_vacation
```
Comment (anoff) [inline lib/data/player.js:100]: What do you think about introducing an optional parameter `ignorePlayerStatus` or something like that.
I feel like the `saveMany` could also provide an improvement for the regular upload from galaxy view, but in that case we want to also overwrite the status fields :D

**Comment by kejith** [comment]
Ist an sich eine sehr gute Idee. Ich weiß nur noch nicht wo genau man dies am besten einbauen könnte. Entweder man übergibt ein Array mit Fields, welche wir schreiben wollen oder wir benutzen einfach einen boolschen Parameter.

Ich werde nun noch schnell den Commit für die Bugs raushauen.

File: lib/data/player.js (hunk 1 above)
Comment (kejith) [reply lib/data/player.js:100]: Is in itself a very good idea. I just don't know where exactly to put this in the best way. Either you pass an array with fields, which we want to write or we just use a boolean parameter.

I will now quickly push out the commit for the bugs.

**Comment by kejith** [comment]
Optional boolean parameter implemented for now. The last two Commits are untested but should be pretty simple and working.

**Pull request merged**